vita3k/gui/include/gui/state.h

```diff
@@ -121,6 +126,11 @@ struct GuiState {
     size_t memory_editor_start = 0;
     size_t memory_editor_count = 0;
 
+    bool isButtonClicked = false;
+    bool isCapturingKeys = false;
+    int oldCapturedKey = 0;
+    int capturedKey = 0;
```

Comment:
Since you've made your own `ControlMenuState` [struct](https://github.com/Vita3K/Vita3K/pull/671/files#diff-8bb169b343989310deb3dce11596b6c5R91), I would put these variables in that struct too. :D

Reply:
Then I will have to pass the struct to handle_events. Is that fine?

Reply:
If `ControlMenuState` [is part of](https://github.com/Vita3K/Vita3K/pull/671/files/d0dad126aedca75d5b9cbc1e447d26785b644fca#diff-8bb169b343989310deb3dce11596b6c5R115) `GuiState` can't you do something like: `gui.controls_menu.is_capturing_keys`?

Reply:
Yes I can whooops